Tasks/Maven/CodeAnalysis/mavensonar.ts

```diff
@@ -0,0 +1,112 @@
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" />
```

Comment:
This file consists of  two methods moved here from maventask, and one new method to generate and upload the summary.

The moved methods have had additional arguments added, since they cannot access some variables previously global inside maventask.